### PR TITLE
chore: fix devcontainer on codespaces

### PR DESCRIPTION
**Pull request by felipecrs**
## Changes

This fixes the permissions of /tmp, which for some reason is broken on GitHub Codespaces (but not locally) when the container is created.

1777 is the proper permissions set for /tmp: https://unix.stackexchange.com/a/71625/427940

I still don't know the root cause, and I'll try to investigate.

## Context

- https://github.com/renovatebot/renovate/pull/23378#issuecomment-1637828169

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by felipecrs** [comment]
Ok, I tried a lot of things, but I still could not figure out what's going on and why this issue only happens with Codespaces.

File: .devcontainer/post-create.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [[ "${CODESPACES}" == true ]]; then
+  echo "Fixing permissions of /tmp for GitHub Codespaces..." >&2
+  sudo chmod 1777 /tmp
```
Comment (viceice) [inline .devcontainer/post-create.sh:7]: I don't think we've sudo available on that image

Comment (felipecrs) [reply .devcontainer/post-create.sh:7]: We have, thanks to the common-utils feature: https://github.com/devcontainers/features/tree/main/src/common-utils#readme

**Review by JamieMagee — DISMISSED** [review]

**Comment by felipecrs** [comment]
@JamieMagee sorry I had just forgot a small cosmetic change.

File: .devcontainer/post-create.sh (hunk 1 above)
Comment (viceice) [reply .devcontainer/post-create.sh:7]: ok.

**Review by viceice — APPROVED** [review]

**Pull request merged**